src/qinterface/rotational.cpp

```diff
@@ -62,6 +64,79 @@ void QInterface::Exp(real1 radians, bitLenInt qubit)
     ApplySingleBit(expIdentity, true, qubit);
 }
 
+void matrix2x2Mul(real1 scalar, complex* left, complex* right, complex* out)
+{
+    std::vector<std::future<void>> futures(4);
```

Comment:
I’m not sure why this is launched async as a future; it seems like it’d take basically the same amount of time to just do the math...

Reply:
The actual math operations are significantly more complicated than the syntax. We do almost the same thing in QFusion.

Reply:
It looks like just four or so multiplies. Am I misreading it?

Reply:
It's complex multiplication, with a scalar factor in front:

A * ((B+iC)*(D+iE) + (F+iG)*(H+iJ))

And the simplification of the complex results adds more operations. We've done this in QFusion, and it's faster with futures.

OpenMP might be better than futures in general, but that's another external library, and a project to change the standard consistently throughout.

Reply:
(I will probably ask this again in the future) and we have measured that it is actually faster to queue a bunch of async ops rather than let the compiler and CPU optimize it?

Reply:
I just tested this, and you're right. I will change it, now. I also tested QFusion, and it seems to be a wash, except QFusion does significantly more work, worst-case, basically this, plus a norm and a branch (in any case) followed by a division by a square root.

So, these cases were not as analogous as I thought.